### ISSUE TITLE
expires_in not returned

### ISSUE THREAD

**Issue by Ndrou** [oauthjs/node-oauth2-server#316]
Hi,

The `expires_in` key is not returned in the result when getting access_token or refresh_token with the version 3.0.0-b2 published on npm.

This seems to be related to this line : 

``` javascript
TokenHandler.prototype.getTokenType = function(model) {
  return new BearerTokenType(model.accessToken, model.accessTokenLifetime, model.refreshToken, model.scope);
};
```

`accessTokenLifetime` can't be present on the model object, because it's an object of class `TokenModel` and this class doesn't implement this property.

Have a good day !

**Comment by maxtruxa** [comment]
Good catch. `accessTokenLifeTime` is supposed to be populated [here](https://github.com/oauthjs/node-oauth2-server/blob/cd7776ce19b4e55a0e5800d78e4186c741711b36/lib/models/token-model.js#L45) when `accessTokenExpiresAt` is present, but this is never the case because [`AuthorizationCodeGrantType.saveToken`](https://github.com/oauthjs/node-oauth2-server/blob/cd7776ce19b4e55a0e5800d78e4186c741711b36/lib/grant-types/authorization-code-grant-type.js#L185) is missing expiry time generation (see #304).

**Comment by Ndrou** [comment]
I see the line you speak about on the branch master
```javascript
  if(this.accessTokenExpiresAt) {
    this.accessTokenLifetime = Math.floor((this.accessTokenExpiresAt - new Date()) / 1000);
  }
```

But this line is not present on the npm package. And when I try to use directly the branch master on my project, i have an error with an other part.

Is there a release planned soon?

**Comment by maxtruxa** [comment]
The published version on npm is v2, master is on v3. Those versions are not compatible.

I'm not one of the maintainers, so I can't tell you when v3 is being published. If you want to track the release status you should probably follow #313. In the meantime you could use [my fork](https://github.com/maxtruxa/node-oauth-server) that has (almost) all pending PRs for v3 merged in. I'll create a PR that fixes the problem with missing expiry times some time this week and directly merge it into my fork.

**Comment by Ndrou** [comment]
Yes i've seen this issue. 
I hope this project will progress. 
Looking on npm, I realized that there's a lot of forks of the project.

Thanks @maxtruxa for your answer.

I will use probably your fork until the original project evolves.

**Comment by maxtruxa** [comment]
This was fixed when #318 got merged.
